var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -19,38 +21,39 @@ class Geant4(CMakePackage):
 
     version('10.04', 'b84beeb756821d0c61f7c6c93a2b83de')
     version('10.03.p03', 'ccae9fd18e3908be78784dc207f2d73b')
-    version('10.02.p03', '2b887e66f0d41174016160707662a77b')
-    version('10.02.p02', '6aae1d0fc743b0edc358c5c8fbe48657')
-    version('10.02.p01', 'b81f7082a15f6a34b720b6f15c6289cfe4ddbbbdcef0dc52719f71fac95f7f1c')
-    version('10.01.p03', '4fb4175cc0dabcd517443fbdccd97439')
 
     variant('qt', default=False, description='Enable Qt support')
     variant('vecgeom', default=False, description='Enable vecgeom support')
-    variant('cxx11', default=True, description='Enable CXX11 support')
-    variant('cxx14', default=False, description='Enable CXX14 support')
     variant('opengl', default=False, description='Optional OpenGL support')
     variant('x11', default=False, description='Optional X11 support')
     variant('motif', default=False, description='Optional motif support')
     variant('threads', default=True, description='Build with multithreading')
+    variant('data', default=True, description='Install geant4 data')
 
-    depends_on('cmake@3.5:', type='build')
+    variant('cxxstd',
+            default='11',
+            values=('11', '14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
 
-    conflicts('+cxx14', when='+cxx11')
-    conflicts('+cxx11', when='+cxx14')
+    depends_on('cmake@3.5:', type='build')
 
     # C++11 support
-    depends_on("clhep@2.4.0.0+cxx11~cxx14", when="@10.04+cxx11~cxx14")
-    depends_on("clhep@2.3.4.3+cxx11~cxx14", when="@10.03.p03+cxx11~cxx14")
-    depends_on("clhep@2.3.1.1+cxx11~cxx14", when="@10.02.p01+cxx11~cxx14")
-    depends_on("clhep@2.3.1.1+cxx11~cxx14", when="@10.02.p01+cxx11~cxx14")
-    depends_on("clhep@2.2.0.4+cxx11~cxx14", when="@10.01.p03+cxx11~cxx14")
+    depends_on("clhep@2.4.0.0 cxxstd=11", when="@10.04 cxxstd=11")
+    depends_on("clhep@2.3.4.6 cxxstd=11", when="@10.03.p03 cxxstd=11")
+    depends_on("vecgeom cxxstd=11", when="+vecgeom cxxstd=11")
 
     # C++14 support
-    depends_on("clhep@2.4.0.0+cxx11~cxx14", when="@10.04~cxx11+cxx14")
-    depends_on("clhep@2.3.4.3+cxx11~cxx14", when="@10.03.p03~cxx11+cxx14")
-    depends_on("clhep@2.3.1.1~cxx11+cxx14", when="@10.02.p02~cxx11+cxx14")
-    depends_on("clhep@2.3.1.1~cxx11+cxx14", when="@10.02.p01~cxx11+cxx14")
-    depends_on("clhep@2.2.0.4~cxx11+cxx14", when="@10.01.p03~cxx11+cxx14")
+    depends_on("clhep@2.4.0.0 cxxstd=14", when="@10.04 cxxstd=14")
+    depends_on("clhep@2.3.4.6 cxxstd=14", when="@10.03.p03 cxxstd=14")
+    depends_on("vecgeom cxxstd=14", when="+vecgeom cxxstd=14")
+
+    # C++17 support
+    patch('cxx17.patch', when='@:10.03.p99 cxxstd=17')
+    patch('cxx17_geant4_10_0.patch', level=1, when='@10.04.00: cxxstd=17')
+    depends_on("clhep@2.4.0.0 cxxstd=17", when="@10.04 cxxstd=17")
+    depends_on("clhep@2.3.4.6 cxxstd=17", when="@10.03.p03 cxxstd=17")
+    depends_on("vecgeom cxxstd=17", when="+vecgeom cxxstd=17")
```

Comment:
With #10561 merged, and `xerces-c` always on, `depends_on('xerces-c cxxstd=NN', when='cxxstd=NN')` should be added for 11, 14 and 17.